[yuanchengg] iP

yuanchengg opened the pull request:
# WinnerBot

> "A Winner never Loses." - WinnerBot

WinnerBot frees your mind of having to remember things you need to do. It's :
- text-based
- easy to learn
- ~~FAST~~ **_SUPER_** _FAST_ to use

All you need to do is, 
1. download it from [here]{http://www.jibble.org/stfw/).
2. double-click it
3. add your tasks
4. let it manage your tasks for you 👍 👍 👍 

And it is **_FREE_**!!!

Features:
- [X] Managing tasks
- [ ] Reminders (coming soon)
- [ ] Powered by AI (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
deleted Duke.java

Commit:
created new class Winner.java with greet and exit

Commit:
level 0

Commit:
level 0 with horizontal lines

Commit:
level 1 - echo commands, exits when bye, indentation included

Commit:
level 2 - can add task and added arraylist to keep track of task

Commit:
level 2 - fixed break after printing list of tasks

Commit:
scanner close

Commit:
created Task.java

Commit:
added "mark" command and changed equalsIgnoreCase to matches (?i)

Commit:
fixed bug with no spacing

Commit:
better printed souts and used repeat for line, fixed bug for getting task number from input

Commit:
add taskCount to keep track of total number of task and edit what chatbot prints when task is added

Commit:
add ToDo class with overridden addTaskToString and taskToString methods

Commit:
add Deadline class with overridden addTaskToString and taskToString methods

Commit:
add Event class with overridden addTaskToString and taskToString methods

Commit:
add support for tracking three types of tasks - ToDos, Deadlines, Events

Commit:
update what is printed when mark/unmark is called to include the description for respective Task

Commit:
update printing such that uppercases are converted to lowercases in sout and edit input.txt, EXPECTED.TXT to do automated text UI testing

Commit:
create WinnerException.java class to handle errors specific to Winner

Commit:
include exceptions to handle errors for commands

Commit:
add deleteTask() method to print which task is deleted when called

Commit:
add support for deleting tasks from the list and update input matches to allow inputs with spaces/slight typos in front or behind of keywords

Commit:
fix mark and unmark task error

Commit:
Java coding standard: edit line breaks

Line breaks after operator.

Let's edit line breaks to be before an operator, improving readability.

Commit:
Save tasks in hard disk and load tasks from hard disk

Tasks are not saved after chat with bot ends.

Let's create new methods to see if a file exists to store the task list,
and create one if it does not exist. We read tasks and load them into
the chatbot. Changes to the task list are written to the file and stored
once conversation ends.

Commit:
Deadline.java: Store deadline dates and time

Store dates and time as String.

Teach chatbot to store dates and time as LocalDateTime.

Commit:
Winner.java: Update Deadline usage

Update exception messages to ask user to print in the correct format.

Handle accepting dates in a certain format and print in a different
format.

Update loadTasks and saveTasks to handle loading and saving tasks with
LocalDateTime dates and time

Commit:
More OOP

Add new classes to make the code more OOP.

Let's extract out closely related code as classes.

Commit:
More OOP

Allow Winner chatbot to work again

Commit:
Edit display for list method.

Commit:
divide classes into packages

Commit:
test branching

Commit:
Merge branch 'master' into branch-A-Packages

* master:
  test branching

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
Add build.gradle

Add gradle to automate project build

Let's set up gradle so I can build and run Winner using gradle.

Commit:
Add JUnit

Let's add JUnit to automate testing of code.

Commit:
Create JAR file

Allow for easy distribution of app.

Let's package the app as an executable JAR file so that it can be
distributed easily.

Commit:
Add checkstyle.xml and suppressions.xml

Check that code adheres to coding standard.

checkstyle.xml: Contains the set of code style rules to follow.
suppressions.xml: Contains which rules to suppress under in which files.

Let's task checkstyleMain and checkstyleTest to check if the main code
and test code complies with the checkstyle rules, respectively.

Commit:
Update coding standard errors

Fix errors not compliant with coding standards.

Commit:
Add JavaDoc comments

Add JavaDoc comments to all classes.

Let's document classes and methods.

Commit:
Add findTasksWithKeyword() method

Let's add a method to allow users to find their tasks which contain a
specific keyword in the task description.

Commit:
Merge branch 'master' into branch-A-JavaDoc

* master:
  Add findTasksWithKeyword() method

Commit:
TaskList.java: Add JavaDoc comment

Add JavaDoc comment for findTasksWithKeyword() method.

Commit:
Resolve merge conflicts.

Merge branch 'master' into A-CodingStandard

* master:
  TaskList.java: Add JavaDoc comment
  Add findTasksWithKeyword() method
  Add JavaDoc comments

# Conflicts:
#	src/main/java/winner/Deadline.java
#	src/main/java/winner/Event.java
#	src/main/java/winner/Task.java
#	src/main/java/winner/ToDo.java
#	src/main/java/winner/Winner.java
#	src/main/java/winner/WinnerException.java

Commit:
Parser.java: update JavaDoc

Unclear JavaDoc comment for parseInput() method.

Update @throws part in JavaDoc comment for parseInput() method.

Commit:
Update syntax to be in line with coding standards.

Imports are not in the correct order in WinnerTest.java.

Trailing whitespace line in error message in TaskList.java

Let's remove the trailing whitespace line and re-order the imports in
the respective classes to adhere to the coding standards.

zyng233 commented on src/main/java/winner/Deadline.java:
```diff
@@ -0,0 +1,100 @@
+        if (isDone) {
+            return "[D] [X] " + description + " (by: " + formattedDeadline() + ")";
+        }
+        return "[D] [ ] " + description + " (by: " + formattedDeadline() + ")";
```
```suggestion
        return "[D] [ ] " + description + " (by: " + formattedDeadline() + ")";
```

zyng233 replied:
can simplify the logic using ternary operator

zyng233 commented on src/main/java/winner/Deadline.java:
```diff
@@ -0,0 +1,100 @@
+                    + " ".repeat(5) + "[D] [X] " + description + " (by: " + formattedDeadline() + ")";
+        }
+        return super.deleteTask() + "\n"
+                + " ".repeat(5) + "[D] [ ] " + description + " (by: " + formattedDeadline() + ")";
```
```suggestion
                + " ".repeat(5) + "[D] [ ] " + description + " (by: " + formattedDeadline() + ")";
```

zyng233 replied:
can make a toString of the task

zyng233 commented on src/main/java/winner/Event.java:
```diff
@@ -0,0 +1,73 @@
+    @Override
+    public String unmarkDone() {
+        return super.unmarkDone() + "\n"
+                + " ".repeat(5) + "[E] [ ] " + description + " (" + start + " - " + end + ")";
```
```suggestion
                + " ".repeat(5) + "[E] [ ] " + description + " (" + start + " - " + end + ")";
```

zyng233 replied:
can factor the task string to reduce duplication

zyng233 commented on src/main/java/winner/ToDo.java:
```diff
@@ -0,0 +1,63 @@
+    @Override
+    public String taskToString() {
+        if (isDone) {
+            return "[T] [X] " + description;
```
```suggestion
            return "[T] [X] " + description;
```

zyng233 replied:
can make the [T] as a constant

zyng233 reviewed (commented):
Overall it LGTM!!!

Commit:
Add a GUI to the chatbot

Use JavaFX technology to implement the GUI.

Commit:
Fix GUI and display Hello message

Commit:
Fix display of number of tasks in list error

Commit:
Remove CLI WinnerTaskBot features

Commit:
Add Javadoc comments for methods in Winner.java

Commit:
Add assertions

Current classes do not have assertions.

The lack of assertions cause undetected bugs and unclear assumptions.

Let's add assertions to parts of the code where assumptions about the
state of variables or logic flow should always hold true.

This enforces assumptions and help catch potential bugs early.

Commit:
Add WinnerTaskBotCLI.java

Commit:
Update code syntax and Javadoc comments

Code has missing Javadoc comments and multiple syntax errors.

This cause harder maintenance and worse readability.

Add Javadoc comments and fixing syntax errors allows better code
readability.

Let's add Javadoc comment for WinnerTaskBotCli class and remove trailing
whitespaces in Ui.java and Main.java.

This creates a consistent style for the codebase.

Commit:
Refactor Parser.java

Parser.java has some repeat lines of code doing the same thing:
* extraction of task number from user input via regex
* check if task number is empty and throw an exception if it is
* convert String task number to int

This causes code duplication.

Refactor code by extracting method.

Let's remove duplicate lines of code and move them into getTaskNumber
method.

yuanchengg committed:
Merge pull request #3 from yuanchengg/branch-A-CodeQuality

Improve code quality

Commit:
Merge branch 'master' into branch-A-Assertions

* master:
  Refactor Parser.java
  Update code syntax and Javadoc comments
  Add WinnerTaskBotCLI.java
  Add Javadoc comments for methods in Winner.java

yuanchengg committed:
Merge pull request #2 from yuanchengg/branch-A-Assertions

Add assertions

Commit:
Add help command to provide access to help page for users

Commit:
Add Javadoc comments

Commit:
Update Javadoc comments

Commit:
Add CommandType Enum class

Commit:
Add Command class

Commit:
Add Javadoc comments

Commit:
Fix code quality

Commit:
Update README

Commit:
Update README

Commit:
Update README
